packages/compass-connections/src/connection-scoped-global-app-registry.ts

```diff
@@ -0,0 +1,66 @@
+import type { ConnectionInfo } from '@mongodb-js/connection-info';
+import {
+  type AppRegistry,
+  createServiceLocator,
+  useGlobalAppRegistry,
+} from 'hadron-app-registry';
+import { useConnectionInfo } from '@mongodb-js/connection-storage/provider';
+import { useMemo } from 'react';
+
+type ConnectionScopedGlobalAppRegistryEvents =
+  | 'schema-analyzed'
+  | 'documents-refreshed'
+  | 'documents-paginated'
+  | 'document-inserted';
```

Comment:
I don't think we want compass-connection to be the source of truth for the event types, similarly how we don't want app-registry package to be the source of truth for all the other events. Using this should be just a intermediate step in the direction of removing these events completely anyway, so this should probably be more generic

Reply:
@kmruiz I think your question is related to what Sergey asked here so replying both here. I typed the events explicitly to keep a track of what all events have been moved to our ConnectionScopedGlobalAppRegistry and also because I had this thought of using maybe some lint to warn about usage of `globalAppRegistry.emit`.

But because we want to get rid of this type of interaction, I agree that it would be fine to just have it as a string.

Reply:
If there is a way (please pardon the usage of forbidden words) to mark `ConnectionScopedGlobalAppRegistry.emit` as `@Deprecated` or something similar, like in Java/Kotlin/Scala so we can see warnings on the usage of this and easily track them later, I'm totally in.

Reply:
There is the jsdoc `@deprecated` but we will require some kind of lint integration to report on it. I guess I will just put that in with a comment for the usage of `ConnectionScopedAppRegistry.emit`?